prov/usnic/src/usdf_av.c

```diff
@@ -352,12 +354,12 @@ usdf_am_insert_sync(struct fid_av *fav, const void *addr, size_t count,
 		}
 		if (ret == 0) {
 			dest->ds_dest = *u_dest;
-			free(u_dest);
 			fi_addr[i] = (fi_addr_t)dest;
 			++ret_count;
 		} else {
 			fi_addr[i] = FI_ADDR_NOTAVAIL;
 		}
+		free(u_dest);
```

Comment:
I don't think this is right.  I think you need to `u_dest = NULL;` at the top of each loop iteration to make this code correct.  It may work fine right now if `usd_create_dest` is always assigning to `u_dest` for us, but I wouldn't count on that normally.